### PR TITLE
feat(tree): add none selection mode

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #3121

## Summary

This introduces a new `none` selection mode that will still allow tree navigation, but prevent an item from being selected. The `calciteTreeSelect` event will emit as well after an item is 'selected'.

### CODE REVIEW

**Comment by jcfranco** [comment]
@macandcheese Could you test the latest tweaks? 🙇

**Review by macandcheese — APPROVED** [review]
✅

File: src/components/tree-item/tree-item.scss [hunk 1]
```diff
@@ -177,7 +183,7 @@
 }
 
 // dropdown expanded and not selected
-:host([has-children][expanded]:not([selected])) > .node-container {
+:host([has-children][expanded]:not([selected]):not([selection-mode="none"])) > .node-container {
```
Comment (driskull) [inline src/components/tree-item/tree-item.scss:186]: nitpick: some of these selectors are getting pretty complex. Maybe at some point we move this logic to the tsx file?

**Review by driskull — APPROVED** [review]
LGTM!

File: src/components/tree-item/tree-item.scss (hunk 1 above)
Comment (jcfranco) [reply src/components/tree-item/tree-item.scss:186]: Yeah, I went with the simplest option for now. I think we'll have to do a pass on complex selectors and see which ones are better off as dynamic classes vs. their styling equivalent. cc @benelan

**Pull request merged**